src/derive/arc.rs

```diff
@@ -0,0 +1,191 @@
+use syn::parse_quote;
+use syn::spanned::Spanned;
+
+use crate::utils::deref_expr;
+use crate::utils::generics_declaration_to_generics;
+use crate::utils::signature_to_method_call;
+use crate::utils::trait_to_generic_ident;
+
+pub fn derive(trait_: &syn::ItemTrait) -> syn::Result<syn::ItemImpl> {
+    // build the methods
+    let mut methods: Vec<syn::ImplItemMethod> = Vec::new();
+    for item in trait_.items.iter() {
+        if let syn::TraitItem::Method(ref m) = item {
+            if let Some(receiver) = m.sig.receiver() {
+                match receiver {
+                    syn::FnArg::Receiver(r) if r.mutability.is_some() => {
+                        let msg = "cannot derive `Rc` for a trait declaring `&mut self` methods";
```

Comment:
Please update the message for `Arc` instead of `Rc` here !

Reply:
ah, missed those. Will do. I will rebase the other PR's.